File: Problem9/euler.cpp
#include <cmath>
#include <cstdio>
#include <vector>
#include <iostream>
#include <algorithm>
using namespace std;


int main() {
    int testcases, n, k, highest, temp;
    cin>>testcases;
    while(testcases--)
    {
        highest = -1;
        cin>>n;
        for(int i =1; i<n; i++)
            for(int j =i; j<n; j++)
            {
                k = n - i -j;
                if(k>j)
                {
                    if(i*i + j*j == k*k)
                    {
                        temp = i *j * k;
                        if(temp>highest)
                        {
                            highest = temp;
                        }
                    }
                }
            }
        cout<<highest<<endl;
    }
    return 0;
}

File: Problem9/euler.py
for testcases in range(int(raw_input())):
    highest = -1
    n = int(raw_input())
    for i in range(1, n/3):
        for j in range(i, n/2):
            k = n - i - j
            if k > j:
                if i**2 + j**2 == k**2:
                    temp = i*j*k
                    if temp > highest:
                        highest = temp
    print highest